zkvm/src/txlog.rs

```diff
@@ -76,6 +98,84 @@ impl TxID {
         entry.commit_to_transcript(&mut t);
         t.challenge_bytes(b"merkle.leaf", result);
     }
+
+    /// Builds a precomputed Merkle tree for the entries.
+    pub fn build_tree(list: &[Entry]) -> (Self, MerkleRoot) {
+        let t = Transcript::new(b"ZkVM.txid");
+        let root = Self::build_node(t, list);
+        (
+            TxID(root.node),
```

Comment:
i don't think it's necessary to return txid here — we can simply return MerkleTree object and then query its `.root` field or method. Also let's not tie it to txid too much — we are likely to factor this out for other use in the blockchain headers.

Reply:
That said, it's fine to use `&[Entry]`, but later it'll be something like `&[I] where I: MerkleItem`